src/stashd/runtime.rs

```diff
@@ -223,19 +223,27 @@ impl Runtime {
         Ok(Reply::Schema(schema))
     }
 
-    fn rpc_consign(
+    // TODO: Rename into `transfer` (since it will produce both Consignment and
+    //       Disclosure)
+    fn rpc_transfer(
         &mut self,
-        request: &ConsignRequest,
+        request: &TransferRequest,
     ) -> Result<Reply, ServiceErrorDomain> {
         debug!("Got CONSIGN {}", request);
 
-        let mut transitions = request.other_transition_ids.clone();
-        transitions.insert(request.contract_id, request.transition.node_id());
+        let mut transitions = request.other_transitions.clone();
+        transitions.insert(request.contract_id, request.transition.clone());
 
         // Construct anchor
         let mut psbt = request.psbt.clone();
-        let (anchors, map) = Anchor::commit(transitions, &mut psbt)
-            .map_err(|err| ServiceErrorDomain::Anchor(format!("{}", err)))?;
+        let (anchors, map) = Anchor::commit(
+            transitions
+                .iter()
+                .map(|(contract_id, ts)| (*contract_id, ts.node_id()))
+                .collect(),
+            &mut psbt,
+        )
```

Comment:
First, it leaving ids only will make logic more complex here, since we would not be able to join main trainsition with other into homogeneous structure in lines 232...233 above. Next, I assume in the future we may like to do extra data checks here, so better to keep transitions in full around: it's a tiny overload.

Reply:
Oh, btw, we need them to do a disclosure right below: https://github.com/rgb-org/rgb-node/pull/148/commits/f0206bc3b5316851165a28beb3dc693642a2d268